### PR TITLE
Bug 1875668 - Bypass cache when intercepting URL requests

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1875668

### CODE REVIEW

**Comment by rocketsroger** [comment]
With fix:
[Screen_recording_20240226_135145.webm](https://github.com/mozilla-mobile/firefox-android/assets/6579541/ade0728b-2dc1-4408-8ddc-b01b5beeb3f7)

**Review by gabrielluong — APPROVED** [review]
Let's add a test before landing.

File: fenix/app/src/test/java/org/mozilla/fenix/components/UrlRequestInterceptorTest.kt [hunk 1]
```diff
@@ -147,6 +148,28 @@ class UrlRequestInterceptorTest {
         )
     }
 
+    @Test
+    fun `WHEN a Google request end in #ip=1 is loaded THEN request bypass cache`() {
+        val uri = "https://www.google.com#ip=1"
```
Comment (owlishDeveloper) [inline fenix/app/src/test/java/org/mozilla/fenix/components/UrlRequestInterceptorTest.kt:153]: nit: I'd use an actual search URL

**Review by owlishDeveloper — APPROVED** [review]

File: fenix/app/src/test/java/org/mozilla/fenix/components/UrlRequestInterceptorTest.kt (hunk 1 above)
Comment (rocketsroger) [reply fenix/app/src/test/java/org/mozilla/fenix/components/UrlRequestInterceptorTest.kt:153]: Make sense.  I'll update.  Thanks

**Comment by rocketsroger** [comment]
@Mergifyio backport releases_v124

**Pull request merged**